File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/Order.cs
namespace OctoSupply.Api.Models;

public sealed record Order(
    int OrderId,
    int BranchId,
    string OrderDate,
    string Name,
    string? Description,
    string Status);

public sealed class CreateOrderRequest
{
    public int BranchId { get; set; }
    public string OrderDate { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Status { get; set; } = "pending";
}

public sealed class UpdateOrderRequest
{
    public int? BranchId { get; set; }
    public string? OrderDate { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? Status { get; set; }
}

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/order.py
from __future__ import annotations

from pydantic import BaseModel


class Order(BaseModel):
    orderId: int
    branchId: int
    orderDate: str
    name: str
    description: str | None = None
    status: str


class CreateOrderRequest(BaseModel):
    branchId: int
    orderDate: str
    name: str
    description: str | None = None
    status: str = "pending"


class UpdateOrderRequest(BaseModel):
    branchId: int | None = None
    orderDate: str | None = None
    name: str | None = None
    description: str | None = None
    status: str | None = None
